sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -1037,20 +1068,54 @@ public BlobClient(Uri blobUri, TokenCredential credential, BlobClientOptions opt
         }
         #endregion Upload
 
-        // NOTE: TransformContent is no longer called by the new implementation
-        // of parallel upload.  Leaving the virtual stub in for now to avoid
-        // any confusion.  Will need to be added back for encryption work per
-        // #7127.
-
         /// <summary>
-        /// Performs a transform on the data for uploads. It is a no-op by default.
+        /// Applies client-side encryption to the data for upload.
         /// </summary>
-        /// <param name="content">Content to transform.</param>
-        /// <param name="metadata">Content metadata to transform.</param>
+        /// <param name="content">
+        /// Content to encrypt.
+        /// </param>
+        /// <param name="metadata">
+        /// Metadata to add encryption metadata to.
+        /// </param>
+        /// <param name="async">
+        /// Whether to perform this operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Cancellation token.
+        /// </param>
         /// <returns>Transformed content stream and metadata.</returns>
-        internal virtual (Stream, Metadata) TransformContent(Stream content, Metadata metadata)
+        private async Task<(Stream, Metadata)> ClientSideEncryptInternal(
```

Comment:
Maybe we could keep the `TransformContent` idea and still inject encryption just make it all internal? that way we'd keep crypto related stuff in other class/namespace without cluttering blobclient.

Reply:
I've pulled out all the crypto transformations, but have left in the explicit call to an encryption method instead of a generic TransformContent. Upcoming push.